source/git-api.js

```diff
@@ -128,7 +128,11 @@ exports.registerApi = function(env) {
   });
 
   app.post(exports.pathPrefix + '/init', ensureAuthenticated, ensurePathExists, function(req, res) {
-    git('init' + (req.param('bare') ? ' --bare --shared' : ''), req.param('path'))
+    var arg = ['init'];
+    if (req.param('bare')) {
+      arg = arg.concat(['--bare', '--shared']);
```

Comment:
You can just use `arg.push('--bare', '--shared');`. No need to create two new arrays.